internal/creds/creds.go

```diff
@@ -42,3 +45,14 @@ func GetCredentialProviderFromInstalledArtifacts(artifacts *tracker.InstalledArt
 	}
 	return "", fmt.Errorf("no credential process found in installed artifacts")
 }
+
+func ValidateCredentialProvider(provider CredentialProvider) error {
```

Comment:
lets break this out and merge it as a separate PR, seems like a good change regardless of rocky.

Reply:
ubuntu 2004 has the same issue